CoverageReport doesn't support embedding second project as subdir without coverage

When embedding a submodule with `add_subdir` from CMake, `enable_coverage_report` fails in the submodule, if the parent has already called it, even when not building in coverage mode, as `clean-coverage` target gets added again.

```
CMake Error at /usr/share/cmake/CoverageReport/EnableCoverageReport.cmake:194 (ADD_CUSTOM_TARGET):
  ADD_CUSTOM_TARGET cannot create target "clean-coverage" because another
  target with the same name already exists.  The existing target is a custom
  target created in source directory
```

--- a/src/CoverageReport/EnableCoverageReport.cmake
+++ b/src/CoverageReport/EnableCoverageReport.cmake
@@ -187,10 +187,11 @@ FUNCTION(ENABLE_COVERAGE_REPORT)
                               DEPENDS ${GLOBAL_DEPENDS})
         ENDIF()
                           
-    ENDIF()
+        # This gets rid of any stale .gcda files. Run this if running a binary
+        # causes lots of messages about about a "merge mismatch for summaries".
+        ADD_CUSTOM_TARGET(clean-coverage
+          COMMAND find ${CMAKE_BINARY_DIR} -name '*.gcda' | xargs rm -f)
 
-    # This gets rid of any stale .gcda files. Run this if a running a binary causes lots of
-    # messages about about a "merge mismatch for summaries".
-    ADD_CUSTOM_TARGET(clean-coverage COMMAND find ${CMAKE_BINARY_DIR} -name '*.gcda' | xargs rm -f)
+    ENDIF()
 
 ENDFUNCTION()

